### DISCUSSION TITLE
Adding text to PDF file

### DISCUSSION THREAD

**Discussion by mohitrobo** [pymupdf/PyMuPDF#1833] [Looking for help] [answered]
I'm trying to add text to an empty PDF file within a textbox using .insert_textbox(). The width of the box(rectangle) is 495 (x_0=50, x_1=545). The point from which the text is added is (x_0,y_0)=(50,50). Here's the code:
![insert_textbox](https://user-images.githubusercontent.com/37827176/180706734-2d40d5eb-3b79-45ba-a176-d377cad5ecd9.jpg)
While for height=5, the text gets pasted onto the page, but for height<5, no text is pasted. How do I find the height of the textbox?

**Comment by JorjMcKie** [comment, marked as answer]
First you must find an **_approximate_** minimum number of lines that will be used for a given box width. Width being 495 we have `math.ceil(text_length / 495)` is 3.
Then multiply this with the "natural" line height of your font, and again use `math.ceil` for the result. The natural line height of a font is `font.ascender - font.descender`. For Times-Roman this is ca. 1.334, so we have
```python
text_length = fitz.get_text_length(text,fontname="tiro")*10
math.ceil(math.ceil(text_length/width)*lineheight)
5
```

You can also overwrite the font lineheight and e.g. take 1.2.

I called the above "approximate" because the algorithm will start a new line whenever a word will not completely fit in the current line. The quotient textlength/width ignores this and will tend to compute a value smaller than the final number of lines.

**Reply by mohitrobo** [reply]
Thank you @JorjMcKie
